### PR TITLE
Extracted the webpack and build into local webpack config

### PR DESCRIPTION
**Pull request by enejb**
#### Changes proposed in this Pull Request

* Extract the valuable webpack config object from calypso's webpack.config.js and merger it with the build script that is coming from jetpack-blocks package.

#### Testing instructions
* Run 
    ```
    npx lerna run clean
    npm ci
    npx lerna bootstrap --scope='@automattic/jetpack-blocks'
    npx lerna run prepublishOnly --stream --scope='@automattic/jetpack-blocks'
    ```

* copy the files over to your local docker `_inc/blocks`  folder:
    ```
    rsync -a --delete packages/jetpack-blocks/dist/ ../../jetpack/_inc/blocks/
    ```

* does everything still work?

### CODE REVIEW

File: packages/jetpack-blocks/webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+			extensions: [ '.json', '.js', '.jsx' ],
+			modules: [ 'node_modules' ],
+		},
+		node: false, // look into ?
+		plugins: _.compact( [
+			new webpack.DefinePlugin( {
+				'process.env.NODE_ENV': JSON.stringify( bundleEnv ),
+				global: 'window',
+			} ),
+			new webpack.IgnorePlugin( /^props$/ ), // look into ?
```
Comment (sirreal) [inline packages/jetpack-blocks/webpack.config.js]: This must be stale, isolating removal in https://github.com/Automattic/wp-calypso/pull/31525

File: packages/jetpack-blocks/webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+const cacheIdentifier = require( '../../server/bundler/babel/babel-loader-cache-identifier' ); // FIX ME
+const config = require( '../../server/config' ); // FIX ME
+// const { workerCount } = require( './webpack.common' ); // todo: shard...
+//
+/**
+ * Internal variables
+ */
+const bundleEnv = config( 'env' );
+const isDevelopment = process.env.DEV_CYCLES === 'true';
+const shouldMinify = false;
```
Comment (sirreal) [inline packages/jetpack-blocks/webpack.config.js]: **Blocker**: We don't want to lose minification!

```suggestion
const shouldMinify = isDevelopment;
```

Seems redundant now…

File: packages/jetpack-blocks/webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+		plugins: _.compact( [
+			new webpack.DefinePlugin( {
+				'process.env.NODE_ENV': JSON.stringify( bundleEnv ),
+				global: 'window',
+			} ),
+			new webpack.IgnorePlugin( /^props$/ ), // look into ?
+			new webpack.IgnorePlugin( /^\.\/locale$/, /moment$/ ),
+			...SassConfig.plugins( { cssFilename, minify: ! isDevelopment } ),
+			new DuplicatePackageCheckerPlugin(),
+			shouldCheckForCycles &&
```
Comment (sirreal) [inline packages/jetpack-blocks/webpack.config.js]: Let's drop this

File: packages/jetpack-blocks/webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+/**
+ * Internal variables
+ */
+const bundleEnv = config( 'env' );
+const isDevelopment = process.env.DEV_CYCLES === 'true';
+const shouldMinify = false;
+// process.env.MINIFY_JS === 'true' ||
+// ( process.env.MINIFY_JS !== 'false' && bundleEnv === 'production' );
+
+const shouldCheckForCycles = process.env.CHECK_CYCLES === 'true';
```
Comment (sirreal) [inline packages/jetpack-blocks/webpack.config.js]: Let's drop this.

```suggestion
```

File: packages/jetpack-blocks/webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+ */
+const cacheIdentifier = require( '../../server/bundler/babel/babel-loader-cache-identifier' ); // FIX ME
+const config = require( '../../server/config' ); // FIX ME
+// const { workerCount } = require( './webpack.common' ); // todo: shard...
+//
+/**
+ * Internal variables
+ */
+const bundleEnv = config( 'env' );
+const isDevelopment = process.env.DEV_CYCLES === 'true';
```
Comment (sirreal) [inline packages/jetpack-blocks/webpack.config.js]: `DEV_CYCLES`? This looks off. Maybe:

```suggestion
const isDevelopment = process.env.NODE_ENV !== 'production';
```

**Review by sirreal — DISMISSED** [review]
Good start, I've given this an initial pass and have suggested some changes.

File: packages/jetpack-blocks/webpack.config.js (hunk 1 above)
Comment (sirreal) [reply packages/jetpack-blocks/webpack.config.js]: Confirmed and removed from main webpack config. Let's remove it.

File: packages/jetpack-blocks/webpack.config.js (hunk 3 above)
Comment (lezama) [reply packages/jetpack-blocks/webpack.config.js]: Done

**Comment by lezama** [comment]
Things seems to build on a second try 👍 

got this error the first time I tried:
```ModuleNotFoundError: Module not found: Error: Can't resolve 'swiper/dist/css/swiper.css' in '~/dev/wp-calypso/packages/jetpack-blocks/src/blocks/slideshow'```

File: packages/jetpack-blocks/.gitignore [hunk 6]
```diff
@@ -0,0 +1,2 @@
+build/*
+dist/*
```
[the remark this answers is not in the document]
Comment (lezama) [reply packages/jetpack-blocks/.gitignore]: https://github.com/Automattic/wp-calypso/pull/31503/commits/b427f4b9e841504c7b0f3d21bfe709ce5047266b

File: packages/jetpack-blocks/webpack.config.js [hunk 7]
```diff
@@ -0,0 +1,185 @@
+/** @format */
```
[the remark this answers is not in the document]
Comment (lezama) [reply packages/jetpack-blocks/webpack.config.js]:
```suggestion
```

File: packages/jetpack-blocks/bin/build.js [hunk 8]
```diff
@@ -6,107 +6,14 @@ process.env.TARGET_BROWSER = 'true';
...
-		},
-		externals: [ ...baseConfig.externals, 'lodash' ],
-		resolve: {
-			...baseConfig.resolve,
-			modules: [ 'node_modules' ],
-		},
-	};
-};
-
 function build() {
```
[the remark this answers is not in the document]
Comment (lezama) [reply packages/jetpack-blocks/bin/build.js]: watch mode?

[the remark this answers is not in the document]
Comment (sirreal) [reply packages/jetpack-blocks/bin/build.js]: `webpack-cli` probably does it better than we do:

https://webpack.js.org/api/cli/#watch-options

[the remark this answers is not in the document]
Comment (lezama) [reply packages/jetpack-blocks/bin/build.js]: 👍 let's erase it

File: packages/jetpack-blocks/webpack.config.js [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+					exclude: /node_modules\//,
+				} ),
+				SassConfig.loader( {
+					preserveCssCustomProperties,
+					includePaths: [ path.join( __dirname, 'client' ) ],
+					prelude: `@import '${ path.join(
+						__dirname,
+						'..',
+						'..',
+						'assets/stylesheets/shared/_utils.scss' // FIX ME
```
Comment (ockham) [inline packages/jetpack-blocks/webpack.config.js]: @enejb and I were discussing that we need to get rid of this implicit dependency on Calypso here. One option we discussed was to move this (and probably a bunch of dependent files) to a new `calypso-style-utils` package.

Comment (ockham) [reply packages/jetpack-blocks/webpack.config.js]: Oh, there's already an issue for this: #31507

Sorry for the noise then :sweat_smile:

File: packages/jetpack-blocks/babel.config.js [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+				corejs: false, // we polyfill so we don't need core-js
+				helpers: true,
+				regenerator: false,
+				useESModules: false,
+			},
+		],
+	] ),
+	overrides: [
+		{
+			test: [ './client/gutenberg/extensions', './packages/jetpack-blocks' ],
```
Comment (ockham) [inline packages/jetpack-blocks/babel.config.js]: These relative paths won't work here, but we do (kinda) need these plugins (long story -- it probably works right now because it's using some fallback JSX transform from the React preset instead, but we shouldn't rely on that).

This will be solved by #31464. For now, let's maybe just change this line to 

```suggestion
			test: [ '.src' ],
```

File: packages/jetpack-blocks/babel.config.js [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+				useBuiltIns: 'entry',
+				// allows es7 features like Promise.prototype.finally
+				shippedProposals: true,
+				// Exclude transforms that make all code slower, see https://github.com/facebook/create-react-app/pull/5278
+				exclude: [ 'transform-typeof-symbol' ],
+			},
+		],
+		'@babel/react',
+	],
+	plugins: _.compact( [
```
Comment (ockham) [inline packages/jetpack-blocks/babel.config.js]: We can drop the `_.compact()` -- not falsey array items to filter here :slightly_smiling_face:

File: packages/jetpack-blocks/babel.config.js [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+					{
+						dir: 'build/i18n-calypso/',
+						headers: {
+							'content-type': 'text/plain; charset=UTF-8',
+							'x-generator': 'calypso',
+						},
+					},
+				],
+			],
+		},
```
Comment (ockham) [inline packages/jetpack-blocks/babel.config.js]: I think we can drop the `build_pot` env here since we're not using i18n tools from within here.

File: packages/jetpack-blocks/package.json [hunk 13]
```diff
@@ -50,7 +54,7 @@
   },
   "scripts": {
     "clean": "npx rimraf dist",
-    "build": "node bin/build.js",
+    "build": "node $NODE_ARGS ./node_modules/webpack/bin/webpack.js --config ./webpack.config.js",
```
Comment (ockham) [inline packages/jetpack-blocks/package.json]: :heart:

Comment (sirreal) [reply packages/jetpack-blocks/package.json]: `webpack-cli` should be a `devDependency`, it's bin will be available, then we should be able to do this:

```suggestion
    "build": "webpack --config ./webpack.config.js",
```

File: packages/jetpack-blocks/webpack.config.js [hunk 14]
```diff
@@ -0,0 +1,155 @@
...
+					workerCount,
+					configFile: path.join( __dirname, 'babel.config.js' ),
+					cacheDirectory: path.join( __dirname, '.cache' ),
+					cacheIdentifier,
+					exclude: /node_modules\//,
+				} ),
+				SassConfig.loader( {
+					preserveCssCustomProperties,
+					includePaths: [ path.join( __dirname, 'client' ) ],
+					prelude: `@import '~@automattic/calypso-color-schemes/src/shared/colors';`,
```
Comment (ockham) [inline packages/jetpack-blocks/webpack.config.js]: :heart:

**Review by ockham — APPROVED** [review]
This is looking good! Left a few minor suggestions -- I think [this one](https://github.com/Automattic/wp-calypso/pull/31503/files#r266662630) might be the most relevant one.

Since this is contained to the `jetpack-blocks` directory and blocks seem to be working well, I think it's okay to merge this! :shipit: 

(There are a few more `FIXME`s in here that we'll need to sort out before actually moving the code over to the JP repo, but we can sort those out in subsequent PRs.)

File: packages/jetpack-blocks/package.json (hunk 13 above)
Comment (ockham) [reply packages/jetpack-blocks/package.json]: Even :heart: er

Comment (lezama) [reply packages/jetpack-blocks/package.json]: https://github.com/Automattic/wp-calypso/pull/31503/commits/ad8acb90f3239b993ad79804fe3409ff5e2ae98f

File: packages/jetpack-blocks/babel.config.js (hunk 12 above)
Comment (lezama) [reply packages/jetpack-blocks/babel.config.js]: https://github.com/Automattic/wp-calypso/pull/31503/commits/384ab7421e1b54d4188ca187ce6a58ea83cb5420

File: packages/jetpack-blocks/babel.config.js (hunk 11 above)
Comment (lezama) [reply packages/jetpack-blocks/babel.config.js]: https://github.com/Automattic/wp-calypso/pull/31503/commits/d544f2b3f5ff756688995033d544fd6d7f549e1b

**Comment by enejb** [comment]
I tested this locally and I was able to build the dist folder as expected via the command outlined in the test.

File: packages/jetpack-blocks/package.json [hunk 15]
```diff
@@ -26,7 +26,9 @@
     "@wordpress/token-list": "^1.1.0",
     "@wordpress/url": "^2.3.3",
     "classnames": "^2.2.6",
+    "circular-dependency-plugin": "5.0.2",
```
Comment (enejb) [inline packages/jetpack-blocks/package.json]: Can we remove this now?

File: packages/jetpack-blocks/.gitignore [hunk 16]
```diff
@@ -0,0 +1,2 @@
+/.cache/
+/dist/
```
Comment (simison) [inline packages/jetpack-blocks/.gitignore]: Looks like this is already covered by root `.gitignore`:

https://github.com/Automattic/wp-calypso/blob/2b2ca2b01466b33add52c533cb6383cef7080be2/.gitignore#L69-L70

File: packages/jetpack-blocks/babel.config.js [hunk 17]
```diff
@@ -0,0 +1,60 @@
+const config = {
+	presets: [
+		[
+			'@babel/env',
+			{
+				targets: { browsers: [ 'last 2 versions', 'Safari >= 10', 'iOS >= 10', 'ie >= 11' ] },
```
Comment (simison) [inline packages/jetpack-blocks/babel.config.js]: Watch for https://github.com/Automattic/wp-calypso/pull/31536 in case this needs rebasing.

File: packages/jetpack-blocks/package.json [hunk 18]
```diff
@@ -37,20 +39,23 @@
...
+    "@automattic/calypso-color-schemes": "file:../calypso-color-schemes",
     "chalk": "^2.4.2",
     "copy-webpack-plugin": "^4.6.0",
     "enzyme": "^3.9.0",
     "enzyme-adapter-react-16": "^1.9.1",
     "enzyme-to-json": "^3.3.5",
     "react": "^16.8.3",
-    "webpack": "^4.29.6"
+    "webpack": "^4.29.6",
+    "webpack-cli": "3.2.3"
```
Comment (simison) [inline packages/jetpack-blocks/package.json]: Prolly needs a `^` here.

Comment (simison) [reply packages/jetpack-blocks/package.json]: Actually there's a mixture of strict and `^` here. I think we'll need `/.packages/.npmrc` to override default "strict" setting from the root config.

File: packages/jetpack-blocks/webpack.config.js (hunk 14 above)
Comment (sirreal) [reply packages/jetpack-blocks/webpack.config.js]: Is there a reason for backtick <code>`</code> quotes here?
```suggestion
					prelude: '@import "~@automattic/calypso-color-schemes/src/shared/colors";',
```

File: packages/jetpack-blocks/webpack.config.js [hunk 19]
```diff
@@ -0,0 +1,155 @@
...
+				} ),
+				FileConfig.loader(),
+			],
+		},
+		resolve: {
+			extensions: [ '.json', '.js', '.jsx' ],
```
Comment (sirreal) [inline packages/jetpack-blocks/webpack.config.js]: For the record, I want the `jsx` extension to go away 🙂 
(no change for this PR)

File: packages/jetpack-blocks/webpack.config.js (hunk 14 above)
Comment (simison) [reply packages/jetpack-blocks/webpack.config.js]: Should we move this import to each file that depends on these colours?

Not necessarily in this PR tho.

File: packages/jetpack-blocks/webpack.config.js [hunk 20]
```diff
@@ -0,0 +1,155 @@
...
+			modules: [ 'node_modules' ],
+		},
+		node: false, // look into ?
+		plugins: _.compact( [
+			new webpack.DefinePlugin( {
+				'process.env.NODE_ENV': JSON.stringify( bundleEnv ),
```
Comment (sirreal) [inline packages/jetpack-blocks/webpack.config.js]: We can likely remove this config dependency:
```suggestion
				'process.env.NODE_ENV': JSON.stringify( process.env.NODE_ENV ),
```

**Review by sirreal — COMMENTED** [review]
Coming along nicely, a few more notes 🙂

File: packages/jetpack-blocks/.gitignore (hunk 16 above)
Comment (simison) [reply packages/jetpack-blocks/.gitignore]: https://github.com/Automattic/wp-calypso/pull/31503/commits/b27473f0a65c04955b3d183bb341295c418d919f

File: packages/jetpack-blocks/package.json (hunk 18 above)
Comment (simison) [reply packages/jetpack-blocks/package.json]: https://github.com/Automattic/wp-calypso/pull/31503/commits/08b8be260d1b12ea564296c97040b2c1383d4d63

Comment (simison) [reply packages/jetpack-blocks/package.json]: >I think we'll need /.packages/.npmrc to override default "strict" setting from the root config.

Nope, not possible. :-( https://docs.npmjs.com/files/npmrc#files

File: packages/jetpack-blocks/webpack.config.js (hunk 20 above)
Comment (simison) [reply packages/jetpack-blocks/webpack.config.js]: https://github.com/Automattic/wp-calypso/pull/31503/commits/7a0909255d2d6dabf6ad4c005fb3366781f4068c

**Comment by simison** [comment]
Desktop tests are failing with:
```
Cloning into '.'...
Warning: Permanently added the RSA host key for IP address '192.30.253.112' to the list of known hosts.

remote: Enumerating objects: 3596, done.        
remote: Total 3596 (delta 0), reused 0 (delta 0), pack-reused 3596        
Receiving objects: 100% (3596/3596), 16.07 MiB | 0 bytes/s, done.
Resolving deltas: 100% (2070/2070), done.
fatal: Couldn't find remote ref tests/add/webpack-jetpack-blocks
Exited with code 128
```
What's going on? 🤔

File: packages/jetpack-blocks/package.json (hunk 15 above)
Comment (simison) [reply packages/jetpack-blocks/package.json]: Removed in https://github.com/Automattic/wp-calypso/pull/31503/commits/da7d36d56f2862d38451315dcd79f2735b44227c

**Comment by sirreal** [comment]
> Desktop tests are failing with:

Intermittent failure. Green again 👍

File: packages/jetpack-blocks/webpack.config.js [hunk 21]
```diff
@@ -0,0 +1,153 @@
...
+		editorSetup,
+		...blockScripts( 'editor', path.join( __dirname, 'src' ), allPresetBlocks ),
+	];
+
+	const webpackConfig = {
+		bail: ! isDevelopment,
+		context: __dirname,
+		entry: {
+			editor: editorScript,
+			...( editorBetaScript && { 'editor-beta': editorBetaScript } ),
```
Comment (sirreal) [inline packages/jetpack-blocks/webpack.config.js]: We expect to always have an `editor-beta` now:
```suggestion
			'editor-beta': editorBetaScript,
```

Comment (simison) [reply packages/jetpack-blocks/webpack.config.js]: Yep, we'll always want to build beta files, even when beta list is empty.

File: packages/jetpack-blocks/webpack.config.js [hunk 22]
```diff
@@ -0,0 +1,153 @@
...
+		node: false, // look into ?
+		plugins: _.compact( [
+			new webpack.DefinePlugin( {
+				'process.env.NODE_ENV': JSON.stringify( process.env.NODE_ENV ),
+				global: 'window',
+			} ),
+			new webpack.IgnorePlugin( /^\.\/locale$/, /moment$/ ),
+			...SassConfig.plugins( { cssFilename, minify: ! isDevelopment } ),
+			new DuplicatePackageCheckerPlugin(),
+			fs.existsSync( presetPath ) &&
```
Comment (simison) [inline packages/jetpack-blocks/webpack.config.js]: This file should always exist now.

File: packages/jetpack-blocks/webpack.config.js [hunk 23]
```diff
@@ -0,0 +1,153 @@
...
+					prelude: '@import "~@automattic/calypso-color-schemes/src/shared/colors";',
+				} ),
+				FileConfig.loader(),
+			],
+		},
+		resolve: {
+			extensions: [ '.json', '.js', '.jsx' ],
+			modules: [ 'node_modules' ],
+		},
+		node: false, // look into ?
```
Comment (simison) [inline packages/jetpack-blocks/webpack.config.js]:
```js
// look into ?
```

Flagging for visibility. :-)

File: packages/jetpack-blocks/webpack.config.js (hunk 22 above)
Comment (simison) [reply packages/jetpack-blocks/webpack.config.js]: https://github.com/Automattic/wp-calypso/pull/31503/commits/d6bfe0656f68ee39266797a5fa28ec816f09cd00

**Review by simison — APPROVED** [review]
Looks good now and builds well, tested in the editor with `JETPACK_BETA_BLOCKS` set `true` & `false`.

File: packages/jetpack-blocks/webpack.config.js (hunk 23 above)
Comment (sirreal) [reply packages/jetpack-blocks/webpack.config.js]: I think we're good:

https://webpack.js.org/configuration/node/#node

https://github.com/Automattic/wp-calypso/blob/05fec274b13cb1a3603a591cd565bdebb4660190/webpack.config.js#L215

```suggestion
		node: false,
```

**Comment by simison** [comment]
This could do with final +1 since there were a bunch of commits by me and Jon after Bernie's ✅ review.

**Comment by simison** [comment]
I see wrong colours in Jetpack sidebar toggle logo:

Screenshot 2019-03-19 at 12 20 37
Screenshot 2019-03-19 at 12 20 26

Colours are fine in `master`:

image

It's from here:

https://github.com/Automattic/wp-calypso/blob/bd2c8ac6d0c223f5fdc5aaa4071db86c3fa98916/packages/jetpack-blocks/src/shared/jetpack-plugin-sidebar.scss#L31-L39

In `master` we get this CSS:

```css
.edit-post-pinned-plugins .components-icon-button:not(.is-toggled) .jetpack-logo__icon-triangle {
    fill: #fff !important;
}
```

In this branch following:
```css
.edit-post-pinned-plugins .components-icon-button:hover .jetpack-logo__icon-triangle, .edit-post-pinned-plugins .components-icon-button:not(.is-toggled) .jetpack-logo__icon-triangle {
    fill: #fff !important;
    fill: var(--color-white) !important;
}
```

We rely quite a bit on these `--color-*` variables in our blocks so I'd expect other places to be broken as well.

**Comment by simison** [comment]
Rebased.

File: packages/jetpack-blocks/package.json [hunk 24]
```diff
@@ -8,49 +8,53 @@
...
-		"chalk": "^2.4.2",
-		"copy-webpack-plugin": "^4.6.0",
-		"enzyme": "^3.9.0",
-		"enzyme-adapter-react-16": "^1.9.1",
-		"enzyme-to-json": "^3.3.5",
-		"react": "^16.8.3",
-		"webpack": "^4.29.6"
+		"@automattic/calypso-build": "file:../calypso-build",
+		"@automattic/calypso-color-schemes": "file:../calypso-color-schemes",
+		"chalk": "2.4.2",
```
Comment (simison) [inline packages/jetpack-blocks/package.json]: I can't see where `chalk` is used?

**Comment by simison** [comment]
I added `package-lock.json` as per comment:

> Mikael i think you want package locks there. Since jetpack-blocks isn’t ref’d from the root, it doesn’t participate in the root npm-shrinkwrap

(via p1552993298168400-slack-team-calypso)

and now it seems like build is failing:

https://circleci.com/gh/Automattic/wp-calypso/221156?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

Locally, running `npx lerna bootstrap --scope='@automattic/jetpack-blocks'` did pass tho.

File: packages/jetpack-blocks/package.json (hunk 24 above)
Comment (sirreal) [reply packages/jetpack-blocks/package.json]: It was in `bin/build.js`
```suggestion
```

Comment (simison) [reply packages/jetpack-blocks/package.json]: Removed. Will need to re package-lock (unless we remove the whole file https://github.com/Automattic/wp-calypso/pull/31503#issuecomment-474333824).

**Review by simison — DISMISSED** [review]
So current PR status:

- [x] ~Colours need to be fixed. see https://github.com/Automattic/wp-calypso/pull/31503#issuecomment-474314700~ fixed in https://github.com/Automattic/wp-calypso/pull/31503/commits/e034860e64f7eb1b980b4fa09f1e24cfeaf07a46
- [x] ~package-lock removed? see https://github.com/Automattic/wp-calypso/pull/31503#issuecomment-474333824~ fixed in https://github.com/Automattic/wp-calypso/pull/31503/commits/432ed69b6234e9a410a01d21415eb89cc53bf5da
- [x] ~figure out if `cacheIdentifier` should be removed; https://github.com/Automattic/wp-calypso/pull/31503#discussion_r266860353~ removed in https://github.com/Automattic/wp-calypso/pull/31503/commits/13ed79a49fdaa6954864669dc8ed99ebc34228d7

File: packages/jetpack-blocks/webpack.config.js [hunk 25]
```diff
@@ -0,0 +1,151 @@
...
+const FileConfig = require( '@automattic/calypso-build/webpack/file-loader' );
+const Minify = require( '@automattic/calypso-build/webpack/minify' );
+const SassConfig = require( '@automattic/calypso-build/webpack/sass' );
+const TranspileConfig = require( '@automattic/calypso-build/webpack/transpile' );
+const wordpressExternals = require( '@automattic/calypso-build/webpack/wordpress-externals' );
+
+/**
+ * Internal dependencies
+ */
+const cacheIdentifier = require( '../../server/bundler/babel/babel-loader-cache-identifier' ); // FIX ME
```
Comment (simison) [inline packages/jetpack-blocks/webpack.config.js]: This can probably be removed but I suppose that can be a separate PR as well.

Convo p1552997442520300-slack-jetpack-gutenberg

File: packages/jetpack-blocks/babel.config.js (hunk 17 above)
Comment (simison) [reply packages/jetpack-blocks/babel.config.js]: In fact, I think we can remove this to let Babel rely on default `browserslist` config provided by the consumer (Jetpack).

File: packages/jetpack-blocks/webpack.config.js (hunk 25 above)
Comment (sirreal) [reply packages/jetpack-blocks/webpack.config.js]: I'm digging into this right now. Will report back…

Comment (sirreal) [reply packages/jetpack-blocks/webpack.config.js]: Let's drop it.

[Babel-loader default](https://github.com/babel/babel-loader/blob/5da4fa0c673474ab14bc829f2cbe03e60e437858/src/index.js#L178-L182):

```js
      const cacheIdentifier = JSON.stringify({
        options,
        "@babel/core": transform.version,
        "@babel/loader": pkg.version,
      });
```

Our implementation:
https://github.com/Automattic/wp-calypso/blob/7a09dd2cf8219188bdc9c3e668cc8c34108169bc/server/bundler/babel/babel-loader-cache-identifier/index.js#L28-L36

It's essentially the same with the addition of our custom babel async transform which we aren't (I hope) bringing with us.

**Review by sirreal — APPROVED** [review]
If it builds and works, let's ship this. It looks good 👍

**Comment by simison** [comment]
Something still failing with the build. :-/

**Review by sirreal — APPROVED** [review]
Tests are all green, let's ship
